<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1024</width>
    <height>780</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Computação Gráfica - Entregas</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout">
    <item>
     <widget class="DisplayFileWidget" name="areaDeDesenho" native="true">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
        <horstretch>1</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
     </widget>
    </item>
    <item>
     <widget class="QScrollArea" name="scrollArea">
      <property name="minimumSize">
       <size>
        <width>270</width>
        <height>0</height>
       </size>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="controlsContainer">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>251</width>
         <height>856</height>
        </rect>
       </property>
       <layout class="QVBoxLayout" name="verticalLayout">
        <item>
         <widget class="QGroupBox" name="groupBox_transform">
          <property name="title">
           <string>1. Transformações do Objeto</string>
          </property>
          <layout class="QFormLayout" name="formLayout">
           <item row="0" column="0">
            <widget class="QLabel" name="label">
             <property name="text">
              <string>Objeto Alvo:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QComboBox" name="comboBox_objetos"/>
           </item>
           <item row="1" column="0" colspan="2">
            <widget class="Line" name="line_3">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>Translação X:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_dx"/>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="label_6">
             <property name="text">
              <string>Translação Y:</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_dy"/>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="label_22">
             <property name="text">
              <string>Translação Z:</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_dz"/>
           </item>
           <item row="5" column="1">
            <widget class="QPushButton" name="pushButton_transladar">
             <property name="text">
              <string>Aplicar Translação</string>
             </property>
            </widget>
           </item>
           <item row="6" column="0">
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>Escala X:</string>
             </property>
            </widget>
           </item>
           <item row="6" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_sx">
             <property name="value">
              <double>1.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="7" column="0">
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>Escala Y:</string>
             </property>
            </widget>
           </item>
           <item row="7" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_sy">
             <property name="value">
              <double>1.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="8" column="0">
            <widget class="QLabel" name="label_23">
             <property name="text">
              <string>Escala Z:</string>
             </property>
            </widget>
           </item>
           <item row="8" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_sz">
             <property name="value">
              <double>1.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="9" column="1">
            <widget class="QPushButton" name="pushButton_escalar">
             <property name="text">
              <string>Aplicar Escala (Pivô: Centro)</string>
             </property>
            </widget>
           </item>
           <item row="10" column="0">
            <widget class="QLabel" name="label_11">
             <property name="text">
              <string>Ângulo (graus):</string>
             </property>
            </widget>
           </item>
           <item row="10" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_angulo">
             <property name="minimum">
              <double>-360.000000000000000</double>
             </property>
             <property name="maximum">
              <double>360.000000000000000</double>
             </property>
            </widget>
           </item>
           <item row="11" column="1">
            <widget class="QPushButton" name="pushButton_rotacionar_X">
             <property name="text">
              <string>Rotacionar em X (Pivô: Centro)</string>
             </property>
            </widget>
           </item>
           <item row="12" column="1">
            <widget class="QPushButton" name="pushButton_rotacionar_Y">
             <property name="text">
              <string>Rotacionar em Y (Pivô: Centro)</string>
             </property>
            </widget>
           </item>
           <item row="13" column="1">
            <widget class="QPushButton" name="pushButton_rotacionar_Z">
             <property name="text">
              <string>Rotacionar em Z (Pivô: Centro)</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_camera">
          <property name="title">
           <string>2. Visualização: Projeção &amp; Viewport</string>
          </property>
          <layout class="QFormLayout" name="formLayout_2">
           <item row="0" column="0" colspan="2">
            <layout class="QHBoxLayout" name="horizontalLayout_radio">
             <item>
              <widget class="QRadioButton" name="radioButton_ortho">
               <property name="text">
                <string>Ortogonal</string>
               </property>
               <property name="checked">
                <bool>true</bool>
               </property>
              </widget>
             </item>
             <item>
              <widget class="QRadioButton" name="radioButton_perspective">
               <property name="text">
                <string>Perspectiva</string>
               </property>
              </widget>
             </item>
            </layout>
           </item>
           <item row="1" column="0" colspan="2">
            <widget class="QLabel" name="label_ortho_params">
             <property name="text">
              <string>Parâmetros da Window (Volume de Visão):</string>
             </property>
            </widget>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>Left:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_ortho_left"/>
           </item>
           <item row="3" column="0">
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>Right:</string>
             </property>
            </widget>
           </item>
           <item row="3" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_ortho_right"/>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="label_13">
             <property name="text">
              <string>Bottom:</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_ortho_bottom"/>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="label_14">
             <property name="text">
              <string>Top:</string>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_ortho_top"/>
           </item>
           <item row="6" column="0">
            <widget class="QLabel" name="label_26">
             <property name="text">
              <string>Near (Z min):</string>
             </property>
            </widget>
           </item>
           <item row="6" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_ortho_near"/>
           </item>
           <item row="7" column="0">
            <widget class="QLabel" name="label_27">
             <property name="text">
              <string>Far (Z max):</string>
             </property>
            </widget>
           </item>
           <item row="7" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_ortho_far"/>
           </item>
           <item row="8" column="0" colspan="2">
            <widget class="Line" name="line_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
            </widget>
           </item>
           <item row="9" column="0" colspan="2">
            <widget class="QLabel" name="label_15">
             <property name="text">
              <string>Parâmetros da Viewport (Tela):</string>
             </property>
            </widget>
           </item>
           <item row="10" column="0">
            <widget class="QLabel" name="label_16">
             <property name="text">
              <string>X Inicial:</string>
             </property>
            </widget>
           </item>
           <item row="10" column="1">
            <widget class="QSpinBox" name="spinBox_vp_x"/>
           </item>
           <item row="11" column="0">
            <widget class="QLabel" name="label_17">
             <property name="text">
              <string>Y Inicial:</string>
             </property>
            </widget>
           </item>
           <item row="11" column="1">
            <widget class="QSpinBox" name="spinBox_vp_y"/>
           </item>
           <item row="12" column="0">
            <widget class="QLabel" name="label_18">
             <property name="text">
              <string>Largura:</string>
             </property>
            </widget>
           </item>
           <item row="12" column="1">
            <widget class="QSpinBox" name="spinBox_vp_largura"/>
           </item>
           <item row="13" column="0">
            <widget class="QLabel" name="label_19">
             <property name="text">
              <string>Altura:</string>
             </property>
            </widget>
           </item>
           <item row="13" column="1">
            <widget class="QSpinBox" name="spinBox_vp_altura"/>
           </item>
           <item row="14" column="1">
            <widget class="QPushButton" name="pushButton_atualizarCamera">
             <property name="text">
              <string>Aplicar Projeção e Viewport</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox_win_transform">
          <property name="title">
           <string>3. Navegação da Câmera</string>
          </property>
          <layout class="QFormLayout" name="formLayout_3">
           <item row="0" column="0">
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>Translação X:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_win_dx"/>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="label_10">
             <property name="text">
              <string>Translação Y:</string>
             </property>
            </widget>
           </item>
           <item row="1" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_win_dy"/>
           </item>
           <item row="2" column="0">
            <widget class="QLabel" name="label_28">
             <property name="text">
              <string>Translação Z:</string>
             </property>
            </widget>
           </item>
           <item row="2" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_win_dz"/>
           </item>
           <item row="3" column="1">
            <widget class="QPushButton" name="pushButton_win_translate">
             <property name="text">
              <string>Mover Câmera</string>
             </property>
            </widget>
           </item>
           <item row="4" column="0">
            <widget class="QLabel" name="label_12">
             <property name="text">
              <string>Escala X:</string>
             </property>
            </widget>
           </item>
           <item row="4" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_win_sx"/>
           </item>
           <item row="5" column="0">
            <widget class="QLabel" name="label_20">
             <property name="text">
              <string>Escala Y:</string>
             </property>
            </widget>
           </item>
           <item row="5" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_win_sy"/>
           </item>
           <item row="6" column="0">
            <widget class="QLabel" name="label_29">
             <property name="text">
              <string>Escala Z:</string>
             </property>
            </widget>
           </item>
           <item row="6" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_win_sz"/>
           </item>
           <item row="7" column="1">
            <widget class="QPushButton" name="pushButton_win_scale">
             <property name="text">
              <string>Zoom Câmera</string>
             </property>
            </widget>
           </item>
           <item row="8" column="0">
            <widget class="QLabel" name="label_21">
             <property name="text">
              <string>Ângulo Giro Z:</string>
             </property>
            </widget>
           </item>
           <item row="8" column="1">
            <widget class="QDoubleSpinBox" name="doubleSpinBox_win_angle"/>
           </item>
           <item row="9" column="1">
            <widget class="QPushButton" name="pushButton_win_rotate">
             <property name="text">
              <string>Girar Câmera em Z</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <spacer name="verticalSpacer">
          <property name="orientation">
           <enum>Qt::Vertical</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>20</width>
            <height>40</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menubar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1024</width>
     <height>21</height>
    </rect>
   </property>
  </widget>
  <widget class="QStatusBar" name="statusbar"/>
 </widget>
 <customwidgets>
  <customwidget>
   <class>DisplayFileWidget</class>
   <extends>QWidget</extends>
   <header>displayfilewidget.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
